src/backend/distributed/planner/multi_physical_planner.c

```diff
@@ -1335,7 +1383,7 @@ ExtractRangeTableId(Node *node)
  * query are mocked RTE_FUNCTION entries, it first translates the RTE's
  * to a form that expandRTE can handle.
  */
-static void
+static Oid
 ExtractColumns(RangeTblEntry *rangeTableEntry, int rangeTableId, List *dependentJobList,
```

Comment:
What about splitting this function into two ?

The first part that finds `callingRTE` goes into a function, returns `callingRTE`, and we call this function seperately.
Also we pass `callingRTE` to `expandRTE` and extract columnNames etc. as below:

```c
static RangeTblEntry *
JoinRangeTableEntry(JoinExpr *joinExpr, List *dependentJobList, List *rangeTableList)
{
    ............
    RangeTblEntry *leftCallingRTE = GetCallingRTE(leftRTE, leftRangeTableId, dependentJobList);
    Oid leftRelId = leftCallingRTE->relid;
	ExtractColumns(leftCallingRTE, leftRangeTableId, dependentJobList, &leftColumnNames, &leftColumnVars);
    ...........
}




ExtractColumns(RangeTblEntry *callingRTE, Oid rangeTableId, List **columnNames, List **columnVars)
{
    expandRTE(callingRTE, rangeTableId, 0, -1, false, columnNames, columnVars);
}
```

Reply:
(we can give a better name to `GetCallingRTE` etc.)

Reply:
+ comment of ExtractColumns should also be updated